app/views/newflow/login_signup/setup_password_form.html.erb

```diff
@@ -0,0 +1,83 @@
+                                            errors: @handler_result&.errors) %>
+
+                <div class="content control-group info-message">
+                    <%= I18n.t(
+                                :"login_signup_form.setup_your_new_password_description",
+                                email: @email,
+                                edit_your_email: link_to(
+                                    I18n.t(:"login_signup_form.edit_your_email"), change_your_email_path
+                                )
+                            ).html_safe
+                    %>
+                </div>
+
+                <div class="content control-group" id="password-field-container">
+                    <%= label_tag :password, I18n.t(:"login_signup_form.password_label"),
+                                class: 'field-label required'
+                    %>
+                    <div class="input-with-tooltip">
+                        <%=
+                                f.text_field(
+                                    :password,
+                                    type: :password,
+                                    placeholder: I18n.t(:"login_signup_form.password_label"),
+                                    class: password_input_tag
+                                )
+                        %>
+
+                        <div class="tooltip">
+                            <h4>Password requirements</h4>
+                            <div>
+                                <i id="password-requirements-checkmark"
+                                    class="fa fa-check-circle">
+                                </i>
+                                <%= @min_passw_length.to_s %> characters minimum
+                            </div>
+                        </div>
+
+                        <span id="show-hide-button">
+                            <span class="toggle-show-hide">SHOW</span>
+                            <span class="toggle-show-hide" style="display: none">HIDE</span>
+                        </span>
+                    </div>
+                </div>
+                <br />
+
+                <div class="content">
+                    <%= f.submit I18n.t(:"login_signup_form.log_in"),
+                                        disable_with: I18n.t(:"login_signup_form.log_in"),
+                                        class: 'primary' %>
+                </div>
+
+            <% end %>
+        <% end %>
+    </div>
+</div>
+
+<script type="text/javascript">
+    var pwd_input = document.getElementsByClassName("<%= password_input_tag %>")[0];
+    var pwd_checkmark = document.getElementById('password-requirements-checkmark');
+
+    pwd_input.oninput = function(event){
+        if (pwd_input.value.length >= <%= @min_passw_length %>) {
+            pwd_checkmark.classList.add("success");
+        } else {
+            pwd_checkmark.classList.remove("success");
+        }
+    }
+</script>
```

Comment:
this JS block seems identical to the one I commented about earlier,  could it be extracted to a partial?